idioms/coercion-arguments.md

```diff
@@ -0,0 +1,112 @@
+# Use borrowed types for arguments
+
+## Description
+
+Using a target of a deref coercion can increase the flexibility of your code when you are deciding which argument type to use for a function argument which may increase its utility as it increases the number of viable inputs for this function.
+For example; using `&str` instead of a `&String`, or `&[T]` in preference of `&Vec<T>`, or even `&T` as opposed to a `&Box<T>`. 
+Not only that, you can avoid layers of indirection for those instances where the owned type already provides a layer of indirection, as it will be illustrated in the following examples. 
+For instance, a `String` has a layer of indirection, so a `&String` will have two layers of indrection.
+We can avoid this by using `&str` instead, and letting `&String` coerce to a `&str` whenever the function is invoked. 
+
+## Example
+
+For this example, we will illustrate some differences for using `&String` as a function argument versus using a `&str`, but the ideas easily apply to using `&Vec<T>` versus using a `&[T]`, or even using a `&T` versus a `&Box<T>`. 
```

Comment:
This is a repetition of line 6. Let's keep this or the other one.

Reply:
updated it already, press "refresh" :-P